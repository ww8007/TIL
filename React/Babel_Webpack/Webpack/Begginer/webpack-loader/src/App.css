.container {
    border: 1px solid blue;
}

.title {
    color: red;
}