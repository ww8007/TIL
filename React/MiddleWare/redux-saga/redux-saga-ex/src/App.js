import React from 'react';
import CounterContainer from './components/CounterContainer';
import SampleContainer from './components/SampleContainer';
const App = () => {
  return <SampleContainer />;
};

export default App;
