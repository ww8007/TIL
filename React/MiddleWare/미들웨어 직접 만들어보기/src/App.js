import React from 'react';
import CounterContainer from './containers/CounterContainer';

const App = () => {
  return <CounterContainer />;
};

export default App;
