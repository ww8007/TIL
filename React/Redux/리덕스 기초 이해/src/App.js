import React from 'react';
import Counter from './components/Counter';
import Todos from './components/Todos';
import CounterContainer from './containers/CounterContainer';

const App = () => {
  return (
    <div>
      <CounterContainer />
      <hr />
      <Todos />
    </div>
  );
};

export default App;
