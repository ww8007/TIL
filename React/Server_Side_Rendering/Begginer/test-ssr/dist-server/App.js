"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = App;

var _react = _interopRequireWildcard(require("react"));

var _Home = _interopRequireDefault(require("./Home"));

var _About = _interopRequireDefault(require("./About"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function App({
  pages
}) {
  const [page, setPage] = (0, _react.useState)(pages);
  (0, _react.useEffect)(() => {
    //  -1-
    window.onpopstate = event => {
      setPage(event.state);
    };
  }, []); //  -2-

  function onChangePage(e) {
    const newPage = e.target.dataset.page;
    window.history.pushState(newPage, '', `/${newPage}`); //  -3-

    setPage(newPage);
  }

  const PageComponent = page === 'home' ? _Home.default : _About.default; // -4-

  return /*#__PURE__*/_react.default.createElement("div", {
    className: "container"
  }, /*#__PURE__*/_react.default.createElement("button", {
    "data-page": "home",
    onClick: onChangePage
  }, "Home"), /*#__PURE__*/_react.default.createElement("button", {
    "data-page": "about",
    onClick: onChangePage
  }, "About"), /*#__PURE__*/_react.default.createElement(PageComponent, null));
}