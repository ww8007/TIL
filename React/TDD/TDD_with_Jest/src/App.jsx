import React, { useEffect } from 'react';

import ListContainer from './ListContainer';

import tasks from '../fixtures/tasks';

import { setTasks } from './actions';

import { useDispatch } from 'react-redux';

export default function App() {
	const dispatch = useDispatch();
	useEffect(() => {
		dispatch(setTasks(tasks));
	}, []);
	return (
		<div>
			<h1>To-do</h1>
			<ListContainer />
		</div>
	);
}
