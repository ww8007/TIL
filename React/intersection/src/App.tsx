import { useEffect, useState, useRef } from "react";

interface ImageGalleryProps {
	images: string[];
}

const App = () => {
	useEffect(() => {
		const rootMargin = "0px 0px 100px 0px";
		const margins = rootMargin.split(" ").map((val) => parseInt(val));

		const topOverlay = document.createElement("div");
		topOverlay.style.position = "fixed";
		topOverlay.style.top = "0";
		topOverlay.style.left = "0";
		topOverlay.style.right = "0";
		topOverlay.style.height = `${margins[0]}px`;
		topOverlay.style.backgroundColor = "rgba(255, 0, 0, 0.3)";

		const bottomOverlay = document.createElement("div");
		bottomOverlay.style.position = "fixed";
		bottomOverlay.style.bottom = "0";
		bottomOverlay.style.left = "0";
		bottomOverlay.style.right = "0";
		bottomOverlay.style.height = `${margins[2]}px`;
		bottomOverlay.style.backgroundColor = "rgba(255, 0, 0, 0.3)";

		document.body.appendChild(topOverlay);
		document.body.appendChild(bottomOverlay);

		return () => {
			document.body.removeChild(topOverlay);
			document.body.removeChild(bottomOverlay);
		};
	}, []);

	const [images] = useState<string[]>([
		"https://picsum.photos/200/300",
		"https://picsum.photos/200/300",
		"https://picsum.photos/200/300",
		"https://picsum.photos/200/300",
		"https://picsum.photos/200/300",
		"https://picsum.photos/200/300",
		"https://picsum.photos/200/300",
		"https://picsum.photos/200/300",
		"https://picsum.photos/200/300",
		"https://picsum.photos/200/300",
		"https://picsum.photos/200/300"
	]);

	return (
		<div id='box'>
			<h1>Image Gallery</h1>
			<ImageGallery images={images} />
			<Box />
			<div
				style={{
					height: "1000px"
				}}
			/>
		</div>
	);
};

export default App;

function ImageGallery({ images }: ImageGalleryProps) {
	// Use browser console and console.log() for debugging
	return (
		<>
			<div
				style={{
					display: "flex",
					flexWrap: "wrap",
					width: "400px",
					marginTop: "400px"
				}}
			>
				{images.map((image, idx) => (
					<LazyImage src={image} alt={`${idx}th image`} key={idx} />
				))}
			</div>
		</>
	);
}

const Box = () => {
	return (
		<div
			style={{
				position: "fixed",
				width: "100%",
				border: "1px solid red",
				height: "calc(100% - 200px)", // 높이를 증가시킵니다.
				top: "100px",
				zIndex: 200
			}}
		/>
	);
};

interface LazyImageProps {
	src: string;
	alt: string;
}

const LazyImage = ({ src, alt }: LazyImageProps) => {
	const [loaded, setLoaded] = useState(false);

	const imageRef = useRef(null);

	useEffect(() => {
		const observer = new IntersectionObserver(
			(entries) => {
				const [entry] = entries;
				if (entry.isIntersecting) {
					setLoaded(true);
				} else {
					setLoaded(false);
				}
			},
			{
				rootMargin: "0px 0px -100px 0px"
			}
		);

		if (imageRef.current) {
			observer.observe(imageRef.current);
		}

		return () => {
			observer.disconnect();
		};
	}, []);

	return (
		<div ref={imageRef}>
			{!!loaded && (
				<img
					src={src}
					alt={alt}
					style={{
						width: "110px",
						height: "110px"
					}}
				/>
			)}
			{!loaded && (
				<div
					style={{
						width: "110px",
						height: "110px"
					}}
				/>
			)}
		</div>
	);
};
