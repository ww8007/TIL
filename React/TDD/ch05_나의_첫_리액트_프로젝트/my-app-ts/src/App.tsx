import React from 'react';
import logo from './logo.svg';
import './App.css';
import styled, { keyframes } from 'styled-components';

const Container = styled.div`
	text-align: center;
`;

const Header = styled.div`
	background-color: #282c34 !important;
	min-height: 100vh;
	display: flex;
	flex-direction: column;
	align-items: center;
	justify-content: center;
	font-size: calc(10px + 2vim);
	color: white;
`;
const spin = keyframes`
	from {
		transform: rotate(0deg);
	}
	to {
		transform: rotate(360deg);
	}
`;
const AppLogo = styled.img`
	height: 40vmin;
	pointer-events: none;

	@media (prefers-reduced-motion: no-preference) {
		animation: ${spin} infinite 20s linear;
	}
`;

const AppLink = styled.a`
	color: #61dafb;
`;

function App() {
	return (
		<Container>
			<Header>
				<AppLogo src={logo} alt="logo" />
				<p>test</p>
				<AppLink
					href="https://reactjs.org"
					target="_blank"
					rel="noopener noreferrer"
				>
					Learn React
				</AppLink>
			</Header>
		</Container>
	);
}

export default App;
