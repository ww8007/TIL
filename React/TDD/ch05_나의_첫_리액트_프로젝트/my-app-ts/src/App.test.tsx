import React from 'react';
import { render, screen } from '@testing-library/react';
import App from './App';

describe('<App />', () => {
	it('renders component correctly', () => {
		const { container } = render(<App />);

		const linkElement = screen.getByText(/learn react/i);
		expect(linkElement).toBeInTheDocument();

		expect(container.getElementsByTagName('p')).toHaveLength(1);
		expect(container.getElementsByTagName('p')[0]).toHaveTextContent('test');
		expect(container).toMatchSnapshot();
	});
});
