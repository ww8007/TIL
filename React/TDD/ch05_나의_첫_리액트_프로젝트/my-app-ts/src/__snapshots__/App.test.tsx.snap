// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders component correctly 1`] = `
<div>
  <div
    class="sc-bdvvtL bNJbpH"
  >
    <div
      class="sc-gsDKAQ iVPoNE"
    >
      <img
        alt="logo"
        class="sc-dkPtRN iiQfJH"
        src="logo.svg"
      />
      <p>
        test
      </p>
      <a
        class="sc-hKwDye ieeKUj"
        href="https://reactjs.org"
        rel="noopener noreferrer"
        target="_blank"
      >
        Learn React
      </a>
    </div>
  </div>
</div>
`;
